docroot/sites/registrar.uiowa.edu/modules/registrar_core/js/academic_calendar.js

```diff
@@ -0,0 +1,209 @@
+(function ($, Drupal, drupalSettings) {
+  'use strict';
+
+  Drupal.behaviors.academicCalendar = {
+    attach: function (context, settings) {
+      once('academicCalendar', '.academic-calendar', context).forEach(function (element) {
+        // Initialize variables to store all events and unique sessions.
+        let allEvents = [];
+        let uniqueSessions = new Set();
+
+        // Cache objects for frequently used elements.
+        const $form = $('#academic-calendar-filter-form', context);
+        const $groupByMonthCheckbox = $('#group-by-month', context);
+        const $showPreviousEventsCheckbox = $('#show-previous-events', context);
```

Comment:
I think these would be slightly faster with standard JS.

Reply:
seems frontend-ish ;)

Reply:
It does. I'm cool with not doing this now, but want to make sure it gets done so if its not done here it should be done in #7854.